plantdescriptionengine/src/main/java/eu/arrowhead/core/plantdescriptionengine/providedservices/pde_monitor/routehandlers/GetAllPlantDescriptions.java

```diff
@@ -0,0 +1,173 @@
+            .name(DIRECTION)
+            .legalValues(ASC, DESC)
+            .defaultValue(ASC)
+            .build();
+
+        final BooleanParameter activeParam = new BooleanParameter.Builder()
+            .name(ACTIVE)
+            .build();
+
+        final List<QueryParameter> acceptedParameters = List.of(pageParam, sortFieldParam, directionParam, activeParam);
+
+        final QueryParamParser parser;
+
+        try {
+            parser = new QueryParamParser(null, acceptedParameters, request);
+        } catch (final ParseError error) {
+            logger.error("Encountered the following error(s) while parsing an HTTP request: " + error.getMessage());
+
+            response
+                .status(HttpStatus.BAD_REQUEST)
+                .body(ErrorMessage.of(error.getMessage()), CodecType.JSON);
+            return Future.success(response);
+
+        }
+
+        List<eu.arrowhead.core.plantdescriptionengine.providedservices.pde_mgmt.dto.PlantDescriptionEntryDto> entries = pdTracker
+            .getEntries();
+
+        final Optional<String> sortField = parser.getValue(sortFieldParam);
+        if (sortField.isPresent()) {
+            final String sortDirection = parser.getRequiredValue(directionParam);
+            final boolean sortAscending = "ASC".equals(sortDirection);
+            switch (sortField.get()) {
+                case ID:
+                    PlantDescriptionEntry.sortById(entries, sortAscending);
+                    break;
+                case CREATED_AT:
+                    PlantDescriptionEntry.sortByCreatedAt(entries, sortAscending);
+                    break;
+                case UPDATED_AT:
+                    PlantDescriptionEntry.sortByUpdatedAt(entries, sortAscending);
+                    break;
+            }
+        }
+
+        final Optional<Integer> page = parser.getValue(pageParam);
+        if (page.isPresent()) {
+            final int itemsPerPage = parser.getRequiredValue(itemPerPageParam);
+            final int from = Math.min(page.get() * itemsPerPage, entries.size());
+            final int to = Math.min(from + itemsPerPage, entries.size());
+            entries = entries.subList(from, to);
+        }
+
+        final Optional<Boolean> active = parser.getValue(activeParam);
+        if (active.isPresent()) {
+            PlantDescriptionEntry.filterByActive(entries, active.get());
+        }
+
+        // Extend each Plant Description Entry with monitor data retrieved from
+        // monitorable services:
+        final List<MonitorPlantDescriptionEntryDto> extendedEntries = new ArrayList<>();
+        for (final PlantDescriptionEntry entry : entries) {
+            extendedEntries.add(DtoUtils.extend(entry, monitorInfo));
+        }
+
+        PlantDescriptionEntryListDto result = new PlantDescriptionEntryListDto.Builder()
+            .data(extendedEntries)
+            .count(extendedEntries.size())
```

Comment:
count should be the total number of entries before paging

Reply:
Fixed